integration_tests/__tests__/failures.test.js

```diff
@@ -56,5 +56,37 @@ test('not throwing Error objects', () => {
 
 test('works with node assert', () => {
   const {stderr} = runJest(dir, ['node_assertion_error.test.js']);
-  expect(normalizeDots(extractSummary(stderr).rest)).toMatchSnapshot();
+  let summary = normalizeDots(extractSummary(stderr).rest);
+
+  // Node 9 started to include the error for `doesNotThrow`
+  // https://github.com/nodejs/node/pull/12167
+  if (Number(process.versions.node.split('.')[0]) >= 9) {
```

Comment:
Not a fan of this, but can live with it. Worth to refactor this test one day though.